addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -253,7 +282,11 @@ private void initialize() throws IOException {
 
     private void sendAndReadMessage(String command) throws IOException {
         sendMessage(new NhcMessageCmd(command));
-        readMessage(this.nhcIn.readLine());
+        BufferedReader reader = this.nhcIn;
+        if (reader == null) {
+            throw new IOException();
```

Comment:
Should you maybe provide an explanation, would `Objects.requireNonNull` also suffice?

Reply:
@martinvw I put this in to avoid my IDE giving a warning for a potential null pointer access after adding null annotations to the class. The reader variable was introduced for that reason. Objects.requireNonNull does not remove that warning. This should normally not happen. What do you suggest?

Reply:
Then let's go for option A, so put a meaningful message in the exception, also apply this to other places  where you use this pattern.